### ISSUE TITLE
Incorrect blob type when fetching the copy status

### ISSUE THREAD

**Issue by trekawek** [Azure/Azurite#3]
Following code uploads a 16MB binary and then copies it on itself:

```java
CloudStorageAccount cloud = CloudStorageAccount.parse("DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;");
CloudBlobContainer container = cloud.createCloudBlobClient().getContainerReference("test");
container.deleteIfExists();
container.create();

for (int i = 0; i < 100; i++) {
    CloudBlockBlob blob = container.getBlockBlobReference("test" + i);
    System.out.print("Uploading: " + blob.getName() + "... ");
    blob.uploadFromByteArray(new byte[16 * 1024 * 1024], 0, 16 * 1024 * 1024);
    System.out.println("Done");

    blob.startCopy(blob);
    CopyStatus status;
    while (true) {
        blob.downloadAttributes();
        status = blob.getCopyState().getStatus();
        if (status != CopyStatus.PENDING) {
            break;
        }
        Thread.sleep(200);
    }
    Assert.assertEquals(CopyStatus.SUCCESS, status);
}
```

Following exception is thrown randomly after a few iterations:

```
com.microsoft.azure.storage.StorageException: Incorrect Blob type, please use the correct Blob type to access a blob on the server. Expected BLOCK_BLOB, actual UNSPECIFIED.

	at com.microsoft.azure.storage.blob.CloudBlob$8.preProcessResponse(CloudBlob.java:1238)
	at com.microsoft.azure.storage.blob.CloudBlob$8.preProcessResponse(CloudBlob.java:1204)
	at com.microsoft.azure.storage.core.ExecutionEngine.executeWithRetry(ExecutionEngine.java:139)
	at com.microsoft.azure.storage.blob.CloudBlob.downloadAttributes(CloudBlob.java:1197)
	at com.microsoft.azure.storage.blob.CloudBlob.downloadAttributes(CloudBlob.java:1164)
	at org.apache.jackrabbit.oak.segment.azure.LoopFailTest.test(LoopFailTest.java:28)
```

**Comment by trekawek** [comment]
I think the problem is caused by this line:

https://github.com/Azure/Azurite/blob/83c561d515b6ee42df8df77abf3422478a634915/lib/core/blob/StorageManager.js#L601

It removes the `entityType` property from the stored blob.

**Comment by arafato** [comment]
Thanks a lot @trekawek ! Great work! Just merged both your PRs into the main branch. Will be released as `v2.6.5` very soon!
